pkg/sdk/validation/manifest/json_remote_implementation.go

```diff
@@ -93,10 +115,106 @@ func (v *RemoteImplementationValidator) Do(ctx context.Context, _ types.Manifest
+			interfacesInputNames = append(interfacesInputNames, inputTypeInstance.Name)
+		}
+	}
+
+	//2. get implementation additional inputs
+	if entity.Spec.AdditionalInput != nil {
+		for name := range entity.Spec.AdditionalInput.Parameters {
+			implAdditionalInput = append(implAdditionalInput, name)
+		}
+		for name := range entity.Spec.AdditionalInput.TypeInstances {
+			implAdditionalInput = append(implAdditionalInput, name)
+		}
+	}
+
+	//3. get inputs from entrypoint workflow template
+	workflow, err := decodeImplArgsToArgoWorkflow(entity.Spec.Action.Args)
+	if err != nil {
+		return ValidationResult{}, errors.Wrap(err, "while decoding Implementation arguments to Argo workflow")
+	}
+	if workflow != nil && workflow.WorkflowSpec != nil {
+		idx, err := argo.GetEntrypointWorkflowIndex(workflow)
+		if err != nil {
+			return ValidationResult{}, errors.Wrap(err, "while getting entrypoint index from workflow")
+		}
+		workflowArtifacts = append(workflowArtifacts, workflow.Templates[idx].Inputs.Artifacts...)
+	}
+
+	//4. verify if the inputs from Implementation and Interface match with Argo workflow artifacts
+	for _, artifact := range workflowArtifacts {
+		existsInInterface := slices.Contains(interfacesInputNames, artifact.Name)
+		existsInAdditionalInput := slices.Contains(implAdditionalInput, artifact.Name)
+
+		if existsInInterface &&
+			artifact.Optional {
+			validationErrs = append(validationErrs, fmt.Errorf("invalid workflow input artifact %q: it shouldn't be optional as it is defined as Interface input", artifact.Name))
+		}
+		if existsInAdditionalInput && !artifact.Optional {
+			validationErrs = append(validationErrs, fmt.Errorf("invalid workflow input artifact %q: it should be optional, as it is defined as Implementation additional input", artifact.Name))
+		}
+		if !existsInInterface && !existsInAdditionalInput {
+			validationErrs = append(validationErrs, fmt.Errorf("unknown workflow input artifact %q: there is no such input neither in Interface input, nor Implementation additional input", artifact.Name))
+		}
+	}
+
+	return ValidationResult{Errors: validationErrs}, nil
+}
+
+func (v *RemoteImplementationValidator) fetchInterfaceInput(ctx context.Context, interfaceRef hubpublicgraphql.InterfaceReference, hub Hub) (hubpublicgraphql.InterfaceInput, error) {
+	iface, err := hub.FindInterfaceRevision(ctx, interfaceRef, public.WithInterfaceRevisionFields(public.InterfaceRevisionInputFields))
+	if err != nil {
+		return hubpublicgraphql.InterfaceInput{}, errors.Wrap(err, "while looking for Interface definition")
+	}
+	if iface == nil {
+		return hubpublicgraphql.InterfaceInput{}, fmt.Errorf("interface %s:%s was not found in Hub", interfaceRef.Path, interfaceRef.Revision)
+	}
+
+	if iface.Spec == nil || iface.Spec.Input == nil {
+		return hubpublicgraphql.InterfaceInput{}, nil
+	}
+
+	return *iface.Spec.Input, nil
+}
+func decodeImplArgsToArgoWorkflow(implArgs map[string]interface{}) (*argo.Workflow, error) {
```

Comment:
As I commented before, please convert it to method

```suggestion
func (v *RemoteImplementationValidator) decodeImplArgsToArgoWorkflow(implArgs map[string]interface{}) (*argo.Workflow, error) {
```

Reply:
Actually, I was not convinced about that because I felt that `decodeImplArgsToArgoWorkflow` can be reused in the future without the need to create a whole object (as the method is more general). But perhaps, it is overthinking. I modified it.  :+1: